homeassistant/components/sabnzbd/sensor.py

```diff
@@ -1,39 +1,120 @@
+    """Describes Sabnzbd sensor entity."""
+
+
+SENSOR_TYPES: tuple[SabnzbdSensorEntityDescription, ...] = (
+    SabnzbdSensorEntityDescription(
+        key="status",
+        name="Status",
+    ),
+    SabnzbdSensorEntityDescription(
+        key="kbpersec",
+        name="Speed",
+        native_unit_of_measurement=DATA_RATE_MEGABYTES_PER_SECOND,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SabnzbdSensorEntityDescription(
+        key="mb",
+        name="Queue",
+        native_unit_of_measurement=DATA_MEGABYTES,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SabnzbdSensorEntityDescription(
+        key="mbleft",
+        name="Left",
+        native_unit_of_measurement=DATA_MEGABYTES,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SabnzbdSensorEntityDescription(
+        key="diskspacetotal1",
+        name="Disk",
+        native_unit_of_measurement=DATA_GIGABYTES,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SabnzbdSensorEntityDescription(
+        key="diskspace1",
+        name="Disk Free",
+        native_unit_of_measurement=DATA_GIGABYTES,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SabnzbdSensorEntityDescription(
+        key="noofslots_total",
+        name="Queue Count",
+        state_class=SensorStateClass.TOTAL,
+    ),
+    SabnzbdSensorEntityDescription(
+        key="day_size",
+        name="Daily Total",
+        native_unit_of_measurement=DATA_GIGABYTES,
+        entity_registry_enabled_default=False,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    SabnzbdSensorEntityDescription(
+        key="week_size",
+        name="Weekly Total",
+        native_unit_of_measurement=DATA_GIGABYTES,
+        entity_registry_enabled_default=False,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    SabnzbdSensorEntityDescription(
+        key="month_size",
+        name="Monthly Total",
+        native_unit_of_measurement=DATA_GIGABYTES,
+        entity_registry_enabled_default=False,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    SabnzbdSensorEntityDescription(
+        key="total_size",
+        name="Total",
+        native_unit_of_measurement=DATA_GIGABYTES,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
 )
 
+SENSOR_KEYS: list[str] = [desc.key for desc in SENSOR_TYPES]
+
 
-async def async_setup_platform(
+async def async_setup_entry(
     hass: HomeAssistant,
-    config: ConfigType,
+    config_entry: ConfigEntry,
     async_add_entities: AddEntitiesCallback,
-    discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
-    """Set up the SABnzbd sensors."""
-    if discovery_info is None:
-        return
+    """Set up a Sabnzbd sensor entry."""
+
+    sab_api = hass.data[DOMAIN][config_entry.entry_id][KEY_API]
+    client_name = hass.data[DOMAIN][config_entry.entry_id][KEY_NAME]
+    sab_api_data = SabnzbdApiData(sab_api)
```

Comment:
We already create an instance of `SabnzbdApiData` during config entry setup in the init module. We shouldn't need more than one instance of that class.